File: programs/LongestWord.java
public class LongestWord {
    public static String longestWord(String test) {

        StringBuilder builder = new StringBuilder();

        String lw = "";
        
        for(int i=0; i<test.length(); i++) {
            char c = test.charAt(i);
            if((c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z')) {
                builder.append(test.charAt(i));
            } else {
                if(lw.length() < builder.length()) {
                    lw = builder.reverse().toString();
                }
                builder.setLength(0);
            }
        }

        if(lw.length() < builder.length()) {
            lw = builder.reverse().toString();
        }

        return lw;
    }
    public static void main(String[] args) {
        String test = "fun&!! time";
        String result = longestWord(test);
        System.out.println("Longest Word Reversed: " + result);

    }
}

File: programs/longest-word.cpp
#include<bits/stdc++.h>
using namespace std;

string longestWord(string sen) {
    string lw = "";
    string test = "";
    for(char c : sen) {
        if((c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z')) {
            test += c;
        } else {
            if(lw.length() < test.length()) {
                lw = test;
            }
            test = "";
        }
    }

    if(lw.length() < test.length()) {
        lw = test;
    }

    reverse(lw.begin(), lw.end());

    return lw;
}

int main(int argc, char const *argv[])
{
    string test = "fun&!! time";
    string result = longestWord(test);

    cout << "Longest word: " << result << endl;

    return 0;
}

